src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -230,6 +269,27 @@ private static boolean disabled(TopLevelItem item) {
 
     @Override
     public <I extends TopLevelItem> Collection<I> orphanedItems(ComputedFolder<I> owner, Collection<I> orphaned, TaskListener listener) throws IOException, InterruptedException {
+        if (abortBuilds) {
+            Queue<Run<?, ?>> abortedBuilds = new LinkedList<>();
+            for (I item: orphaned) {
+                for (Job<?, ?> job: item.getAllJobs()) {
+                    for (Run<?, ?> build: job.getBuilds()) {
+                        Executor executor = build.getExecutor();
+                        if (executor == null) {
+                            continue;
+                        }
+                        executor.interrupt(Result.ABORTED);
+                        abortedBuilds.add(build);
+                    }
+                }
+            }
+            Run<?, ?> abortedBuild;
+            while ((abortedBuild = abortedBuilds.poll()) != null) {
+                while (abortedBuild.isLogUpdated()) {
+                    Thread.sleep(100L);
+                }
+            }
```

Comment:
I guess we are waiting because https://github.com/jenkinsci/cloudbees-folder-plugin/blob/53bf95c744c8a0ff468ef6d2b699796d2186075e/src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java#L318 relies on it? Definitely there should be a time limit waiting for an interrupt to be processed. It is not generally a big deal if we have to skip one of the orphaned projects because it has a build still running; it should just be processed the next time branch indexing runs.

Reply:
Ceiled with a max wait of 1 minute for all aborted builds.